lib/upsert/merge_function/postgresql.rb

```diff
@@ -116,14 +116,18 @@ def server_version
       end
 
       def schema_query
+        table_name_arguments = table_name.is_a?(Array) ? table_name : ["public", table_name]
+        table_name_arguments.unshift("public") if table_name_arguments.length == 1
+        table_name_arguments << quoted_table_name if table_name_arguments.length == 2
```

Comment:
what does this do?

Reply:
I'm cleaning this up (and writing some specs for it) - basically we need to make sure that when we're looking for the unique index (I also renamed the function to be clearer) we select the correct table - so use schema if specified and if not use the default search path.

Ex, we have `public.pets` and `other_schema.pets` - we need to search for a unique index on whichever table we're targeting for the UPSERT.